### PR TITLE
Added missing flags and documentation to 'UserFlags.cs'.

### PR DESCRIPTION
**Pull request by Red-K0**

### CODE REVIEW

**Comment by KubaZ2** [comment]
The names should be PascalCase. Could you also replace spaces before and after the `summary` content with new lines and remove them after casting?

**Comment by Red-K0** [comment]
> The names should be PascalCase. Could you also replace spaces before and after the `summary` content with new lines and remove them after casting?

Yeah, all done.

**Comment by KubaZ2** [comment]
Could you also remove the 0 prefixes from bitfield values? They look odd for me. Also could you please split the values with new lines? I think it would make the code more readable.

**Comment by Red-K0** [comment]
> Could you also remove the 0 prefixes from bitfield values? They look odd for me. Also could you please split the values with new lines? I think it would make the code more readable.

Split and prefixes removed

**Comment by KubaZ2** [comment]
Could you also remove the empty spaces? I think it should be `Name = value` for all fields. I think the unnecessary indentation is useless.

**Comment by Red-K0** [comment]
> Could you also remove the empty spaces? I think it should be `Name = value` for all fields. I think the unnecessary indentation is useless.

Yeah that's more a remnant of how the code originally looked, removing it

File: NetCord/UserFlags.cs [hunk 1]
```diff
@@ -1,22 +1,194 @@
...
+    /// Undocumented and Private. User has SMS recovery for 2FA enabled.
+    /// </summary>
+    MfaSms = (ulong)1 << 4,
+
+    /// <summary>
+    /// Undocumented and Private. User dismissed the Nitro promotion.
+    /// </summary>
+    PremiumPromoDismissed = (ulong)1 << 5,
+
+
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs:35]:
```suggestion
```

**Review by KubaZ2 — CHANGES_REQUESTED** [review]

**Comment by Red-K0** [comment]
Not sure why it didn't go through the first time, but extra space is removed

**Comment by KubaZ2** [comment]
I reminded myself that https://github.com/KubaZ2/NetCord/blob/alpha/NetCord%2FPermissions.cs are formatted with uL suffix instead of casting. Could you follow that convention?

**Comment by Red-K0** [comment]
uL convention added

File: NetCord/UserFlags.cs [hunk 2]
```diff
@@ -1,22 +1,193 @@
...
-    TeamUser = 1 << 10,
-    BugHunterLevel2 = 1 << 14,
-    VerifiedBot = 1 << 16,
-    EarlyVerifiedBotDeveloper = 1 << 17,
-    DiscordCertifiedModerator = 1 << 18,
-    BotHttpInteractions = 1 << 19,
-    Spammer = 1 << 20,
-    ActiveDeveloper = 1 << 22,
+    /// <summary>
+    /// User is a discord employee.
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]:
```suggestion
    /// User is a Discord employee.
```

File: NetCord/UserFlags.cs [hunk 3]
```diff
@@ -1,22 +1,193 @@
...
+    /// </summary>
+    Staff = 1uL << 0,
+
+    /// <summary>
+    /// User is a discord partner.
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]:
```suggestion
    /// User is a Discord partner.
```

File: NetCord/UserFlags.cs [hunk 4]
```diff
@@ -1,22 +1,193 @@
...
+
+    /// <summary>
+    /// Undocumented. User is the SYSTEM account.
+    /// </summary>
+    System = 1uL << 12,
+
+    /// <summary>
+    /// Undocumented and Private. User has unread messages from Discord.
+    /// </summary>
+    Has_Unread_Urgent_Messages = 1uL << 13,
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]:
```suggestion
    HasUnreadUrgentMessages = 1uL << 13,
```

**Review by KubaZ2 — CHANGES_REQUESTED** [review]

**Comment by KubaZ2** [comment]
Could you also fix casing of `private` when it follows `Undocumented and`?

**Comment by Red-K0** [comment]
Sorry for the delay, casings fixed

File: NetCord/UserFlags.cs [hunk 5]
```diff
@@ -1,22 +1,193 @@
...
+
+    /// <summary>
+    /// Undocumented and private. User has SMS recovery for 2FA enabled.
+    /// </summary>
+    MfaSms = 1uL << 4,
+
+    /// <summary>
+    /// Undocumented and private. User dismissed the Nitro promotion.
+    /// </summary>
+    PremiumPromoDismissed = 1uL << 5,
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]:
```suggestion
    PremiumPromotionDismissed = 1uL << 5,
```

File: NetCord/UserFlags.cs [hunk 6]
```diff
@@ -1,22 +1,193 @@
...
+    /// </summary>
+    TeamPseudoUser = 1uL << 10,
+
+    /// <summary>
+    /// Undocumented and private. User has a pending partner/verification application.
+    /// </summary>
+    InternalApplication = 1uL << 11,
+
+    /// <summary>
+    /// Undocumented. User is the SYSTEM account.
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]:
```suggestion
    /// Undocumented. User is an Official Discord System user (part of the urgent message system).
```

File: NetCord/UserFlags.cs [hunk 7]
```diff
@@ -1,22 +1,193 @@
...
+    /// </summary>
+    CertifiedModerator = 1uL << 18,
+
+    /// <summary>
+    /// Undocumented. User is a bot with an interactions endpoint.
+    /// </summary>
+    BotHttpInteractions = 1uL << 19,
+
+    /// <summary>
+    /// Undocumented. User's account is disabled for spam.
```
Comment (KubaZ2) [inline NetCord/UserFlags.cs]: I don't think it means the account is disabled. Other users have spammers' messages hidden by default as if they were blocked. Maybe something changed though.

**Review by KubaZ2 — CHANGES_REQUESTED** [review]

File: NetCord/UserFlags.cs (hunk 7 above)
Comment (Red-K0) [reply NetCord/UserFlags.cs]: No discord can actually just disable you for spam, had it happen to someone I know.

> Discord has disabled your account for spam and/or platform abuse. Our anti-abuse measures may have flagged your account for any of the following behaviors:
> - Sending a large number of direct messages in a short span of time
> - Participating in a server dedicated to spamming Discord
> - Automating your user account or self-botting
> - Taking other actions on Discord — for example, joining a ton of servers — faster than humanly possible
> - Attempting potentially fraudulent activity

Comment (KubaZ2) [reply NetCord/UserFlags.cs]: This is how being a spammer used to look like.
image

Comment (budgetdevv) [reply NetCord/UserFlags.cs]: A spammer can be disabled or vice-versa. But even the non-disabled ones will have the flag, I checked in my own server @KubaZ2

Comment (KubaZ2) [reply NetCord/UserFlags.cs]: thanks, that means that the comment isn't correct

Comment (Red-K0) [reply NetCord/UserFlags.cs]: Yep, question is what to change it to though. I could just leave it at "User's account is disabled."

Comment (budgetdevv) [reply NetCord/UserFlags.cs]: "User account is flagged as spammer"

Comment (Red-K0) [reply NetCord/UserFlags.cs]: Added that instead then

**Pull request merged**